Fix scan unsupported workflow in SQl DB

yux0 opened the pull request:
<!-- Describe what has changed in this PR -->
**What changed?**
1. Fix the result ordering of SQL
2. Fix version histories deserialization

<!-- Tell your future self why have you made these changes -->
**Why?**


<!-- How have you verified this change? Tested locally? Added a unit test? Checked in staging env? -->
**How did you test it?**
Local testing

<!-- Assuming the worst case, what can be broken when deploying this change to production? -->
**Potential risks**

yycptt reviewed (commented):
Can we add some comment for the new contract of list executions?

yycptt reviewed (commented):
It looks like `minUUID` is also used by ListTaskList, will it cause the same issue there?

yux0 commented:
> Can we add some comment for the new contract of list executions?
Added comments about the behavior.

> It looks like `minUUID` is also used by ListTaskList, will it cause the same issue there?

I think so. Added a comment on this API. This API is only used in Scanner.

yycptt commented on common/persistence/sql/sqlTaskManager.go:
```diff
@@ -264,6 +264,9 @@ type taskListPageToken struct {
 
+// ListTaskList lists tasklist from DB
+// DomainID translates into byte array in SQL. The minUUID is not the minimum byte array.
+// This API could return incomplete result set.
```
Shall we create an issue for this? Looks like this is something we need to fix.

yux0 replied:
https://github.com/uber/cadence/issues/3911
